fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/join/JoinCommutative.java

```diff
@@ -15,28 +15,43 @@
 // specific language governing permissions and limitations
 // under the License.
 
-package org.apache.doris.nereids.rules.exploration;
+package org.apache.doris.nereids.rules.exploration.join;
 
 import org.apache.doris.nereids.rules.Rule;
 import org.apache.doris.nereids.rules.RuleType;
-import org.apache.doris.nereids.trees.plans.JoinType;
+import org.apache.doris.nereids.rules.exploration.OneExplorationRuleFactory;
 import org.apache.doris.nereids.trees.plans.Plan;
 import org.apache.doris.nereids.trees.plans.logical.LogicalJoin;
 
 /**
  * rule factory for exchange inner join's children.
  */
 public class JoinCommutative extends OneExplorationRuleFactory {
+    private boolean justApplyInnerOuterCrossJoin = false;
+
+    private final SwapType swapType;
+
+    /**
+     * If param is true, just apply rule in inner/full-outer/cross join.
+     */
+    public JoinCommutative(boolean justApplyInnerOuterCrossJoin) {
+        this.justApplyInnerOuterCrossJoin = justApplyInnerOuterCrossJoin;
+        this.swapType = SwapType.ALL;
```

Comment:
What's this used for ?

Reply:
limit swap space, we should change `JoinReorderContext ` according to it 
- left deep tree -> bottom join and disable l-asscom
- zig-zag ...
- busy...